specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,592 @@
+    [hWnd](ICoreWebView2StagingTextureStream* sender, ICoreWebView2StagingTextureStreamTextureErrorEventArgs* args) {
+      COREWEBVIEW2_TEXTURE_STREAM_ERROR_KIND kind;
+      HRESULT hr = args->get_Kind(&kind);
+      assert(SUCCEEDED(hr));
+      switch (kind)
+      {
+      case COREWEBVIEW2_TEXTURE_STREAM_ERROR_NO_VIDEO_TRACK_STARTED:
+      case COREWEBVIEW2_TEXTURE_STREAM_ERROR_BUFFER_NOT_FOUND:
+      case COREWEBVIEW2_TEXTURE_STREAM_ERROR_BUFFER_IN_USE:
+        // assert(false);
+        break;
+      default:
+        break;
+      }
+      return S_OK;
+    }).Get(), &texture_token));
+
+  // Add allowed origin for registerTextureStream call. 'registerTextureStream'
+  // call from the Javascript will fail if the requested origin is not registered
+  // with AddWebTextureAllowedOrigin.
+  CHECK_FAILURE(webviewTextureStream->AddWebTextureAllowedOrigin(L"https://edge-webscratch"));
+
+  // Registers listener for video streaming from Javascript.
+  EventRegistrationToken post_token;
+  CHECK_FAILURE(webviewTextureStream->add_WebTextureReceived(Callback<ICoreWebView2StagingTextureStreamWebTextureReceivedEventHandler>(
+    [&](ICoreWebView2StagingTextureStream* sender, ICoreWebView2StagingTextureStreamWebTextureReceivedEventArgs* args) {
+      // Javascript send a texture stream.
+      ComPtr<ICoreWebView2StagingWebTexture> texture_received;
+      args->GetWebTexture(&texture_received);
+
+      ULONGLONG timestamp;
+      texture_received->get_Timestamp(&timestamp);
+      HANDLE handle;
+      texture_received->get_Handle(&handle);
+      DrawTextureWithWICBitmap(handle, timestamp);
+
+      return S_OK;
+    }).Get(), &post_token));
+
+  // Register listener of video stream from the Javascript end.
+  EventRegistrationToken stopped_token;
+  CHECK_FAILURE(webviewTextureStream->add_WebTextureStreamStopped(Callback<ICoreWebView2StagingTextureStreamWebTextureStreamStoppedEventHandler>(
+    [&](ICoreWebView2StagingTextureStream* sender, IUnknown* args) {
+
+      return S_OK;
+    }).Get(), &stopped_token));
+}  // CreateTextureStream
+
+HRESULT CallWebView2API(bool createBuffer,
+                        UINT32 width,
+                        UINT32 height,
+                        bool requestAvailableBufer,
+                        bool sendTexture,
+                        ICoreWebView2StagingTexture* send_texture,
+                        UINT64 timestamp,
+                        ICoreWebView2StagingTexture** texture) {
+  if (createBuffer) {
+    // Create shared buffer.
+    return webviewTextureStream->CreateSharedBuffer(width, height, &texture);
+  }
+
+  if (requestAvailableBufer) {
+    return webviewTextureStream->GetAvailableBuffer(&texture);
+  }
+
+  if (sendTexture) {
+    // Notify the renderer for updated texture on the shared buffer.
+    webviewTextureStream->SetBuffer(send_texture, timestamp);
+    webviewTextureStream->Present();
+  }
+}
+```
+
+# API Details

```

Comment:
Missing sample code and MIDL3 for WinRT/C# APIs. Is that intentional? If so, it needs to be mentioned in the document at the top somewhere about why

Reply:
yes, it is intentional.

/// The proposed APIs have dependency on the DirectX and its internal attributes such as
/// adapter LUID so it supports Win32/C++ and C++/WinRT APIs at this time.

Reply:
Then, I will remove C++/WinRT as developers can use the APIs though like UWP DirectX app in the official Microsoft UWP samples (https://learn.microsoft.com/en-us/windows/uwp/gaming/tutorial--create-your-first-uwp-directx-game, https://github.com/microsoft/Windows-universal-samples/tree/main/Samples/Simple3DGameDX)